dags/airflow-log-cleanup.py

```diff
@@ -0,0 +1,253 @@
+                    Check file permissions.\
+                    To re-run the DAG, ensure that the lock file has been \
+                    deleted ("""
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """)."
+                    exit ${REMOVE_LOCK_FILE_EXIT_CODE}
+                fi
+                exit ${DELETE_STMT_EXIT_CODE}
+            fi
+        else
+            echo "WARN: No File(s)/Directory(s) to Delete"
+        fi
+    else
+        echo "WARN: You're opted to skip deleting the File(s)/Directory(s)!!!"
+    fi
+}
+
+
+if [ ! -f """
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """ ]; then
+
+    echo "Lock file not found on this node! \
+    Creating it to prevent collisions..."
+    touch """
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """
+    CREATE_LOCK_FILE_EXIT_CODE=$?
+    if [ "${CREATE_LOCK_FILE_EXIT_CODE}" != "0" ]; then
+        echo "Error creating the lock file. \
+        Check if the airflow user can create files under tmp directory. \
+        Exiting..."
+        exit ${CREATE_LOCK_FILE_EXIT_CODE}
+    fi
+
+    echo ""
+    echo "Running Cleanup Process..."
+
+    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/*/* -type f -mtime \
+     +${MAX_LOG_AGE_IN_DAYS}"
+    DELETE_STMT="${FIND_STATEMENT} -exec rm -f {} \;"
+
+    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
+    CLEANUP_EXIT_CODE=$?
+
+    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/*/* -type d -empty"
+    DELETE_STMT="${FIND_STATEMENT} -prune -exec rm -rf {} \;"
+
+    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
+    CLEANUP_EXIT_CODE=$?
+
+    FIND_STATEMENT="find ${BASE_LOG_FOLDER}/* -type d -empty"
+    DELETE_STMT="${FIND_STATEMENT} -prune -exec rm -rf {} \;"
+
+    cleanup "${FIND_STATEMENT}" "${DELETE_STMT}"
+    CLEANUP_EXIT_CODE=$?
+
+    echo "Finished Running Cleanup Process"
+
+    echo "Deleting lock file..."
+    rm -f """
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """
+    REMOVE_LOCK_FILE_EXIT_CODE=$?
+    if [ "${REMOVE_LOCK_FILE_EXIT_CODE}" != "0" ]; then
+        echo "Error removing the lock file. Check file permissions. To re-run the DAG, ensure that the lock file has been deleted ("""
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """)."
+        exit ${REMOVE_LOCK_FILE_EXIT_CODE}
+    fi
+
+else
+    echo "Another task is already deleting logs on this worker node. \
+    Skipping it!"
+    echo "If you believe you're receiving this message in error, kindly check \
+    if """
+    + str(LOG_CLEANUP_PROCESS_LOCK_FILE)
+    + """ exists and delete it."
+    exit 0
+fi
+
+"""
+)
+
+for log_cleanup_id in range(1, NUMBER_OF_WORKERS + 1):
+
+    for dir_id, directory in enumerate(DIRECTORIES_TO_DELETE):
+
+        log_cleanup_op = BashOperator(
+            task_id="log_cleanup_worker_num_"
+            + str(log_cleanup_id)
+            + "_dir_"
+            + str(dir_id),
+            bash_command=log_cleanup,
+            params={"directory": str(directory), "sleep_time": int(log_cleanup_id) * 3},
```

Comment:
It seems `directory` is passed into the jinja2 template. Do we really need to case it into `str` here? I thought jinja2 will handle that